internal/btf/btf.go

```diff
@@ -176,16 +198,50 @@ func LoadKernelSpec() (*Spec, error) {
 }
 
 func loadKernelSpec() (*Spec, error) {
-	fh, err := os.Open("/sys/kernel/btf/vmlinux")
-	if os.IsNotExist(err) {
-		return nil, fmt.Errorf("can't open kernel BTF at /sys/kernel/btf/vmlinux: %w", ErrNotFound)
-	}
+	release, err := unix.KernelRelease()
 	if err != nil {
-		return nil, fmt.Errorf("can't read kernel BTF: %s", err)
+		return nil, fmt.Errorf("can't read kernel release number: %w", err)
+	}
+
+	// use same list of locations as libbpf
+	// https://github.com/libbpf/libbpf/blob/9a3a42608dbe3731256a5682a125ac1e23bced8f/src/btf.c#L3114-L3122
+	locations := []struct {
+		pathFmt string
+		raw     bool
+	}{
+		/* try canonical vmlinux BTF through sysfs first */
+		{"/sys/kernel/btf/vmlinux", true /* raw BTF */},
+		/* fall back to trying to find vmlinux ELF on disk otherwise */
+		{"/boot/vmlinux-%s", false},
+		{"/lib/modules/%s/vmlinux-%s", false},
+		{"/lib/modules/%s/build/vmlinux", false},
+		{"/usr/lib/modules/%s/kernel/vmlinux", false},
+		{"/usr/lib/debug/boot/vmlinux-%s", false},
+		{"/usr/lib/debug/boot/vmlinux-%s.debug", false},
+		{"/usr/lib/debug/lib/modules/%s/vmlinux", false},
+	}
+
+	var spec *Spec
+	for _, loc := range locations {
+		path := fmt.Sprintf(loc.pathFmt, release)
+		fh, err := os.Open(path)
+		if err != nil {
+			continue
+		}
+		defer fh.Close()
+		if loc.raw {
+			spec, err = loadNakedSpec(fh, internal.NativeEndian, nil, nil)
+		} else {
+			spec, err = loadNakedSpecFromReader(fh, nil, nil)
+		}
+
+		if err != nil {
+			continue
+		}
+		return spec, nil
 	}
-	defer fh.Close()
 
-	return loadNakedSpec(fh, internal.NativeEndian, nil, nil)
+	return nil, fmt.Errorf("can't find kernel BTF: %w", ErrNotSupported)
```

Comment:
Sorry, you were right. ErrNotFound is the correct return value here.

```suggestion
	return nil, fmt.Errorf("BTF for kernel version %s: %w", release, ErrNotFound)
```